pkg/deployer/helm/ensure.go

```diff
@@ -106,14 +110,54 @@ func (h *Helm) ApplyFiles(ctx context.Context, files map[string]string, exports
 		return err
 	}
 
-	h.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseSucceeded
 	h.DeployItem.Status.ProviderStatus = statusData
-	h.DeployItem.Status.ObservedGeneration = h.DeployItem.Generation
 	if err := h.kubeClient.Status().Update(ctx, h.DeployItem); err != nil {
 		h.DeployItem.Status.LastError = lsv1alpha1helper.UpdatedError(h.DeployItem.Status.LastError,
 			currOp, "UpdateStatus", err.Error())
 		return err
 	}
+
+	if h.ProviderConfiguration.HealthChecks.Disable {
+		h.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseSucceeded
+		h.DeployItem.Status.ObservedGeneration = h.DeployItem.Generation
+		h.DeployItem.Status.LastError = nil
+		return nil
+	}
+
+	return h.CheckResourcesHealth(ctx)
+}
+
+// CheckResourcesHealth checks if the managed resources are Ready/Healthy.
+func (h *Helm) CheckResourcesHealth(ctx context.Context) error {
+	var (
+		currOp      = "CheckResourcesHealthHelm"
+		helmDecoder = serializer.NewCodecFactory(Helmscheme).UniversalDecoder()
+		status      = &helm.ProviderStatus{}
+	)
+
+	if _, _, err := helmDecoder.Decode(h.DeployItem.Status.ProviderStatus.Raw, nil, status); err != nil {
+		return err
+	}
+
+	if len(status.ManagedResources) == 0 {
+		return nil
+	}
+
+	objects := make([]*unstructured.Unstructured, len(status.ManagedResources))
+	for i, ref := range status.ManagedResources {
+		obj := kutil.ObjectFromTypedObjectReference(&ref)
+		objects[i] = obj
+	}
+
+	timeOut := time.Duration(h.ProviderConfiguration.HealthChecks.TimeOutSeconds) * time.Second
+	if err := health.WaitObjectsHealthy(ctx, timeOut, h.log, h.kubeClient, objects); err != nil {
```

Comment:
i think it would make sense to have a dedicated resource health condition

Reply:
not sure what you mean here but I hope this would fit in a new PR :smile:

Reply:
yes it also fits in a new PR.
I meant that we have condition on a deploy item (`.status.conditions`). We could also add the health as a condition.
e.g. 
```
status:
  conditions:
  - reason: RESOURCES_HEALTH
    status: true
    message: ...
```